mm2src/coins_activation/src/eth_with_token_activation.rs

```diff
@@ -35,10 +46,7 @@ impl From<EthActivationV2Error> for EnablePlatformCoinWithTokensError {
                 EnablePlatformCoinWithTokensError::InvalidPayload(e)
             },
             EthActivationV2Error::InvalidPathToAddress(e) => EnablePlatformCoinWithTokensError::InvalidPayload(e),
-            #[cfg(target_arch = "wasm32")]
-            EthActivationV2Error::ExpectedRpcChainId => {
-                EnablePlatformCoinWithTokensError::InvalidPayload(err.to_string())
-            },
+            EthActivationV2Error::ChainIdNotSet => EnablePlatformCoinWithTokensError::InvalidPayload(err.to_string()),
```

Comment:
I think 'InvalidPayload' error assumes that something wrong is with the activation request (as 'payload' is usually meant as the content of a message). In case of a missing chain id I'd say this is rather an internal error or invalid config error.

Reply:
Fixed here https://github.com/KomodoPlatform/komodo-defi-framework/pull/1979/commits/22013005986f908bda35bb21b07803d7f798e90a

I also added a checklist item
>  Document changes before merging PRs. e.g. chain_id is needed for all evm coins/tokens. chain_id should be part of platform coin config only actually and retrieved from platform coin.

To remember documenting that chain_id is now mandatory ref. https://github.com/KomodoPlatform/komodo-defi-framework/pull/1979#discussion_r1546841562

Btw, I think `chain_id`, `required_confirmations`, `avg_blocktime`, `derivation_path` should be inherited from platform coin and no need to add them to every token config as it's currently done. There will be backward compatibility issues for v1 activation methods if this is done though, maybe I should open an issue.

Reply:
> Btw, I think chain_id, required_confirmations, avg_blocktime, derivation_path should be inherited from platform coin and no need to add them to every token config as it's currently done. There will be backward compatibility issues for v1 activation methods if this is done though, maybe I should open an issue.

P.S. This can be part of custom token/asset import functionality which we will work on soon.

Reply:
found an issue for this already for QRC20 https://github.com/KomodoPlatform/komodo-defi-framework/issues/756